iset.mm

```diff
@@ -109111,6 +109111,46 @@ infinite descent (here implemented by strong induction).  (Contributed
     TZVNUTUSVHVORUSUTVAVHUGZUSUTVAVHOZBAMUNQVNBGAVNBVQPZVNSVNAVRPZUHQVNBAVSVTVN
     BAVSVTVPUIUOUJUSUTVCVGVHUPVAABUQNUR $.
 
+  $( Lemma for ~ sqrt2irrap .  The square root of 2 is apart from a positive
+     rational expressed as a numerator and denominator.  (Contributed by Jim
+     Kingdon, 2-Oct-2021.) $)
+  sqrt2irraplemnn $p |- ( ( A e. NN /\ B e. NN ) ->
+      ( sqrt ` 2 ) # ( A / B ) ) $=
+    ( cn wcel cdiv co c2 csqrt cfv cap wbr nnred nngt0d ltled mpbird cq syl2anc
+    cc0 wb cz wa cexp simpl nnsqcld 0red simpr nnrpd sqrtdivd sqrtsqd eqtrd wne
+    oveq12d cmul sqne2sq nncnd 2cnd nnap0d divmulap3d necon3bid nnzd 2z zq mp1i
+    znq qapne cr cle qre syl divgt0d 2re a1i 0le2 sqrt11ap syl22anc eqbrtrrd cc
+    nnz sylan qcn sqrt2re recni apsym mpbid ) ACDZBCDZUAZABEFZGHIZJKZWIWHJKZWGA
+    GUBFZBGUBFZEFZHIZWHWIJWGWOWLHIZWMHIZEFWHWGWLWMWGWLWGAWEWFUCZUDZLZWGRWLWGUEZ
+    WTWGWLWSMZNWGWMWGBWEWFUFZUDZUGUHWGWPAWQBEWGAWGAWRLZWGRAXAXEWGAWRMNUIWGBWGBX
+    CLZWGRBXAXFWGBXCMNUIULUJWGWOWIJKZWNGJKZWGXHWNGUKZWGXIWLGWMUMFZUKABUNWGWNGWL
+    XJWGWLGWMWGWLWSUOWGUPWGWMXDUOWGWMXDUQURUSOWGWNPDZGPDZXHXISWGWLTDWMCDXKWGWLW
+    SUTXDWLWMVDQZGTDXLWGVAGVBVCWNGVEQOWGWNVFDZRWNVGKGVFDZRGVGKZXGXHSWGXKXNXMWNV
+    HVIZWGRWNXAXQWGWLWMWTWGWMXDLXBWGWMXDMVJNXOWGVKVLXPWGVMVLWNGVNVOOVPWGWHVQDZW
+    IVQDZWJWKSWGWHPDZXRWEATDWFXTAVRABVDVSWHVTVIXSWGWIWAWBVLWHWIWCQWD $.
+
+  ${
+    $d Q a b $.
+    $( The square root of 2 is irrational.  That is, for any rational number,
+       ` ( sqrt `` 2 ) ` is apart from it.  In the absence of excluded middle,
+       we can distinguish between this and "the square root of 2 is not
```

Comment:
Maybe:
> In the absence of excluded middle, this is strictly stronger than "the square root..."

Reply:
I really struggle with wording on this sort of thing. On the one hand no theorem is stronger than any other theorem in the sense of http://us.metamath.org/mpeuni/2th.html . On the other hand we need to find the language to talk about what sqrt2irr proves versus what sqrt2irrap proves.

Reply:
I think "stronger" is appropriate, here.  It's a common wording, even if of course two truths are equivalent.  You can put quotation marks around it. It says that one implication is more "obvious" than the other.  For a justification, there is probably a weak base theory A (something like IFOL plus some real number axioms) such that over A, sqrt2irrap implies sqrt2irr but not conversely.  Basically, A should be able to prove that apartness implies difference, but should not be so strong that it would prove sqrt2irrap from sqrt2irr.  (If, on the other hand, there is no such theory A, then that is a result in itself!)